include/mega/megaclient.h

```diff
@@ -178,6 +181,31 @@ class MEGA_API FetchNodesStats
     dstime timeToTransfersResumed;
 };
 
+
+// Helper class for MegaClient.  Executes independent tasks on a separate thread or threads.
+// The number of threads can be 0 (eg. for helper MegaApi) in which case something queued is 
+// immediately executed synchronously on the caller's thread
+// Queueing a nullptr instead of a function to execute will cause the threads to exit.
+struct MegaClientAsyncQueue
+{
+    void push(std::function<void(SymmCipher&)> f);
+    void clearQueue();
+
+    MegaClientAsyncQueue(Waiter& w, unsigned threadCount);
+    ~MegaClientAsyncQueue();
+
+private:
+    Waiter& mWaiter;
+    std::mutex mMutex;
+    std::condition_variable mConditionVariable;
+    std::deque<std::function<void(SymmCipher&)>> mQueue;
+    std::vector<std::thread> mAsyncThreads;
```

Comment:
That name is a bit confusing to me. How does an "async thread" differ from a "thread"?

Reply:
Threads for async operations.  Named in consultation with the department of redundancies department ;-).  updated